File: CodeForces-Global-Round-8/d.cpp
#include<bits/stdc++.h>
using namespace std;
#define debug(x) cout<<#x<<" = "<<x<<endl;
 
int main() {
	int n;
	scanf("%d",&n);
	vector <long long> a(n);
	long long m=0;
	int ind=0;
	for (int i=0;i<n;i++) {
		scanf("%lld",&a[i]);
		if (a[i]>m) {
			ind=i;
		}
	}
	sort(a.begin(),a.end());
	for (int i=0;i<n-1;i++) {
		int index=n-1;
		while (index>i) {
			if (a[index]|a[i]>a[index]) {
				long long temp=a[index];
				a[index]|=a[i];
				a[i]&=temp;
			}
			index--;
		}
	}
	long long ans=0;
	for (int i=0;i<n;i++) {
		ans+=(a[i]*a[i]);
	}
	cout<<ans<<endl;
}	

File: CodeForces-Global-Round-8/d.py
n=int(input())
a=[]
m=0
ind=-1
x=list(input().split(' '))
#print(x)
for i in range(n): 
	a.append(int(x[i]))	
	if a[i]>=m:
		ind=i;
a=sorted(a)
print(a)
mi=a[len(a)-1]
for i in range(n):
	if a[i]==mi: 
		break;
	for j in range(i+1,n):
		if (a[i]|a[j])>a[i]:
			temp=a[i]
			a[i]=a[j]|a[i]
			a[j]=a[j]&temp
		if mi<a[j]:
			mi=a[j]	
ans=0
for i in range(n):
	ans+=(a[i]*a[i])
print(ans)
